SNIPS/snip-86.md

```diff
@@ -0,0 +1,601 @@
+          "uniqueItems": true,
+          "errorMessage": "Message must contain min 6-8 unique items "
+        },
+        "StarknetDomain": {
+          "type": "array",
+          "items": {
+            "type": "object",
+            "properties": {
+              "name": {
+                "type": "string",
+                "enum": ["name", "chainId", "version", "revision"],
+                "errorMessage": "Name must be one of 'name', 'chainId', 'version', 'revision'"
+              },
+              "type": {
+                "type": "string",
+                "enum": ["felt", "string"],
+                "errorMessage": "Type must be 'felt' or 'string'"
+              }
+            },
+            "required": ["name", "type"],
+            "additionalProperties": false,
+            "errorMessage": "Items must include name and type"
+          },
+          "minItems": 4,
+          "maxItems": 4,
+          "uniqueItems": true,
+          "errorMessage": "StarknetDomain must contain exactly 4 unique items"
+        }
+      },
+      "required": ["Message", "StarknetDomain"],
+      "additionalProperties": false,
+      "errorMessage": "Types must include Message and StarknetDomain"
+    }
+  },
+  "required": ["domain", "message", "primaryType", "types"],
+  "additionalProperties": false,
+  "errorMessage": "Data must include domain, message, primaryType, types"
+}
+```
+
+#### Message Fields
+
+This specification defines the following SIWS Message fields that can be parsed from a SIWS Message by following the rules in [Structured JSON Message Format](#structured-json-message-format).
+
+- `domain` REQUIRED. An object containing information about the domain requesting the signing.
+
+  - `chainId` REQUIRED. The Starknet network identifier. Its value MUST be either "SN_SEPOLIA" or "SN_MAIN".
+  - `name` REQUIRED. The name of the application requesting the signing.
+  - `version` REQUIRED. The version of the application. Its value MUST be a string in the format x.y.z and cannot exceed 31 characters.
+  - `revision` REQUIRED. The Typed Data revision. Its value MUST be a string and MUST be "1" for this specification.
+
+- `message` REQUIRED. An object containing the core components of the SIWS message.
+
+  - `version` REQUIRED. The version of the SIWS message. Its value MUST be a string in the format x.y.z and cannot exceed 31 characters.
+  - `address` REQUIRED. The Starknet address performing the signing. Its value MUST be a hexadecimal string with 66 characters, including the '0x' prefix.
+  - `statement` REQUIRED. A human-readable ASCII assertion that the user will sign.
+  - `uri` REQUIRED. An RFC 3986 URI referring to the resource that is the subject of the signing.
+  - `nonce` REQUIRED. A random string used to prevent replay attacks, at least 8 alphanumeric characters and no more than 31 characters.
+  - `issuedAt` REQUIRED. The time when the message was generated. Its value MUST be an ISO 8601 datetime string.
```

Comment:
Can you elaborate on the purpose of this field?

Reply:
To confirm, you mean the `issuedAt` field?

Reply:
Yes, sorry 😅

Reply:
2 considerations I had in mind are that:- 

- it can help prevent replay attacks to ensure the message is reused within an appropriate timeframe
- can be used together with `validUntil` and/or `validAfter` to make sure that the message is only valid for a certain duration

Reply:
IMO the field is not needed, the nonce will be enough to prevent replay attacks

Reply:
Open to ideas on this, perhaps the others can chime in here to see if its really needed